File: 0049-group-anagrams/0049-group-anagrams.py

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res=defaultdict(list)
        for i in strs:
            sorted_word=''.join(sorted(i))
            res[sorted_word].append(i)
        return list(res.values())

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> res=new HashMap<>();
        for(String i:strs){
            char[] sortedArr=i.toCharArray();
            Arrays.sort(sortedArr);
            String sortedword=new String(sortedArr);
            res.putIfAbsent(sortedword,new ArrayList<>());
            res.get(sortedword).add(i);
        }
        return new ArrayList<>(res.values());
    }
}